modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ExclusiveParameterModelValidator.java

```diff
@@ -0,0 +1,121 @@
+     * {@inheritDoc}
+     */
+    @Override
+    public void validate(ExtensionModel model) throws IllegalModelDefinitionException
+    {
+        new ExtensionWalker()
+        {
+            @Override
+            public void onOperation(HasOperationModels owner, OperationModel model)
+            {
+                validateExclusiveParameterGroups(model, OPERATION);
+            }
+
+            @Override
+            public void onConfiguration(ConfigurationModel model)
+            {
+                validateExclusiveParameterGroups(model, CONFIG);
+            }
+
+            @Override
+            public void onConnectionProvider(HasConnectionProviderModels owner, ConnectionProviderModel model)
+            {
+                validateExclusiveParameterGroups(model, PROVIDER);
+            }
+
+            @Override
+            public void onSource(HasSourceModels owner, SourceModel model)
+            {
+                validateExclusiveParameterGroups(model, SOURCE);
+            }
+        }.walk(model);
+    }
+
+    /**
+     * @param model         to be validated
+     * @param componentName of the model
+     * @throws IllegalModelDefinitionException if there is a nested {@link ParameterGroup} or
+     * the parameter {@link MetadataType} is not a {@link SimpleType}
+     */
+    private void validateExclusiveParameterGroups(EnrichableModel model, String componentName) throws IllegalModelDefinitionException
+    {
+
+        Optional<ParameterGroupModelProperty> parameterGroupModelProperty = model.getModelProperty(ParameterGroupModelProperty.class);
+        if (parameterGroupModelProperty.isPresent() && parameterGroupModelProperty.get().hasExclusion())
+        {
+            for (ParameterGroup pg : parameterGroupModelProperty.get().getGroups())
+            {
+                Optional<ParameterGroupModelProperty> nestedParameterGroup = pg.getModelProperty(ParameterGroupModelProperty.class);
+                if (nestedParameterGroup.isPresent())
+                {
+                    throw new IllegalModelDefinitionException(format("Parameter group of class '%s' is annotated with '%s' so it cannot contain any nested parameter group on its inside.",
+                                                                     pg.getType().getName(), Exclusion.class.getName(), componentName));
+                }
+
+                pg.getParameters().stream().forEach(f -> getFieldMetadataType(f, typeLoader).accept(new MetadataTypeVisitor()
+                {
+                    @Override
+                    protected void defaultVisit(MetadataType metadataType)
+                    {
+                        throw new IllegalModelDefinitionException(format("Parameter group of class '%s' using exclusion cannot contain any complex type inside but '%s' was found", pg.getType().getName(), f.getType().getName()));
+                    }
+
+                    @Override
+                    public void visitSimpleType(SimpleType simpleType)
```

Comment:
This will fail with things like an `ArrayType`, meaning that you can't have a `List` in your exclusion type. That's not ok, we should be able to have any parameter we want as long as it is not a Pojo.

I'd change this to fail only with `ObjectType` and allow everything else.